File: apps/chat-ios/AmazonChimeSDKMessagingDemo/messaging-websocket/session/MessagingSession.swift
//
//  MessagingSession.swift
//  AmazonChimeSDKMessagingDemo
//
//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
//  SPDX-License-Identifier: Apache-2.0
//
import Foundation

@objc public protocol MessagingSession {
    func start()

    func stop()

    func addMessagingSessionObserver(observer: MessagingSessionObserver)

    func removeMessagingSessionObserver(observer: MessagingSessionObserver)
}

File: apps/chat-android/messaging-websocket/src/main/java/com/amazonaws/services/chime/sdk/messaging/session/MessagingSession.kt
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

package com.amazonaws.services.chime.sdk.messaging.session

interface MessagingSession {
    fun start()

    fun stop()

    fun addMessagingSessionObserver(observer: MessagingSessionObserver)

    fun removeMessagingSessionObserver(observer: MessagingSessionObserver)
}
